### PR TITLE
Skip bpf_spin_lock field

### PR DESCRIPTION
**Pull request by kmateuszssak**

### CODE REVIEW

File: lib/common.c [hunk 1]
```diff
@@ -208,7 +245,7 @@ static int setup_struct_field_descriptor_set_btf(psabpf_btf_t *btf_md, psabpf_st
             }
         }
 
-        const struct btf_type *member_type = psabtf_get_type_by_id(btf_md->btf, md.effective_type_id);
+        member_type = psabtf_get_type_by_id(btf_md->btf, md.effective_type_id);
```
Comment (tatry) [inline lib/common.c:248]: Second type resolution? (see line 219, I think that both are the same if using `md.effective_type_id`)

File: lib/common.c [hunk 2]
```diff
@@ -196,6 +213,26 @@ static int setup_struct_field_descriptor_set_btf(psabpf_btf_t *btf_md, psabpf_st
...
+            fprintf(stderr, "invalid type\n");
+            return false;
+        }
+        const char *type_name = btf__name_by_offset(btf_md->btf, member_type->name_off);
+        if (type_name == NULL) {
+            fprintf(stderr, "invalid type\n");
+            return false;
+        }
+        if (strcmp(type_name, "bpf_spin_lock") == 0)
+            cos = true;
```
[the remark this answers is not in the document]
Comment (kmateuszssak) [reply lib/common.c]: Done

File: lib/common.c (hunk 1 above)
Comment (kmateuszssak) [reply lib/common.c]: Done

**Review by tatry — APPROVED** [review]
LGTM

**Pull request merged**